src/plugins/kibana_usage_collection/server/collectors/event_loop_delays/rollups/integration_tests/daily_rollups.test.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { Logger, ISavedObjectsRepository } from '../../../../../../../core/server';
+import {
+  createTestServers,
+  TestElasticsearchUtils,
+  TestKibanaUtils,
+  createRootWithCorePlugins,
+} from '../../../../../../../core/test_helpers/kbn_server';
+import { rollDailyData } from '../daily';
+import { mocked } from '../../event_loop_delays.mocks';
+
+import {
+  SAVED_OBJECTS_DAILY_TYPE,
+  serializeSavedObjectId,
+  EventLoopDelaysDaily,
+} from '../../saved_objects';
+import moment from 'moment';
+
+const { startES } = createTestServers({
+  adjustTimeout: (t: number) => jest.setTimeout(t),
+});
+let esServer: TestElasticsearchUtils;
+
+function createRawObject(pid: number, date: moment.MomentInput) {
```

Comment:
why do we want to specify different `pid` in the test?

Reply:
Needed for creating raw objects. I've updated the code to make it optional. There is a test case in the file to double check that it deletes all object with different pids.